app/controllers/posts_controller.rb

```diff
@@ -672,7 +673,7 @@ def edit_checks
       redirect_back fallback_location: root_path
     end
 
-    if !@post_type.is_public_editable && !(@post.user == current_user || current_user.is_moderator)
+    if !@post_type.is_public_editable && !(@post.user == current_user || current_user&.at_least_moderator?)
```

Comment:
Untested suggestion:

```suggestion
    if !@post_type.is_public_editable && !@post.user == current_user && current_user&.standard?
```

Without testing, I think this change follows from [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws), but which is more readable is probably personal taste. Personally I find the suggestion easier to parse quickly because it doesn't require parentheses for operator precedence.